### ISSUE TITLE
mk2parts.sh: Drive mount check uses incomplete search string

### ISSUE THREAD

**Issue by irushavm** [jumpnow/meta-rpi64#1]
First of all I want to say that your article regarding building Yocto helped me a lot so thanks for that!

When trying to partition the SD card(/dev/sde) using the `mk2parts.sh` script, the script failed to continue even though the partitions were unmounted:
```bash
sudo ./meta-rpi64/scripts/mk2parts.sh sde
Looks like partitions on device /dev/sde are mounted
Not going to work on a device that is currently in use
cgroup on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)
```
The grep command on line 15 searches for only `sde` which is matched by the word `nsdelegate`.

By using `$DRIVE` instead of `$1`, the mount check was able to be correctly identify if any SD card partitions were mounted.

Diff:
```diff
project meta-rpi64/
diff --git a/scripts/mk2parts.sh b/scripts/mk2parts.sh
index 0a50e15..d2a3c83 100755
--- a/scripts/mk2parts.sh
+++ b/scripts/mk2parts.sh
@@ -12,7 +12,7 @@ else
        exit 1
 fi
 
-mount | grep -q ${1}
+mount | grep -q $DRIVE
```

**Comment by scottellis** [comment]
Thanks. I will fix this.

Best regards,
Scott

"Irusha Vidanamadura" <notifications@github.com> said:

**Comment by scottellis** [comment]
Fixed with commit 76b2b06a3044e7d605e17d7cacfe933773754f1d
